src/wp-includes/rest-api/endpoints/class-wp-rest-attachments-controller.php

```diff
@@ -374,6 +384,220 @@ public function post_process_item_permissions_check( $request ) {
 		return $this->update_item_permissions_check( $request );
 	}
 
+	/**
+	 * Checks if a given request has access to editing media.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function edit_media_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'upload_files' ) ) {
+			return new WP_Error(
+				'rest_cannot_edit_image',
+				__( 'Sorry, you are not allowed to upload media on this site.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return $this->update_item_permissions_check( $request );
+	}
+
+	/**
+	 * Applies edits to a media item and creates a new attachment record.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, WP_Error object on failure.
+	 */
+	public function edit_media_item( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/image.php';
+
+		$attachment_id = $request['id'];
+
+		// This also confirms the attachment is an image.
+		$image_file = wp_get_original_image_path( $attachment_id );
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
```

Comment:
There is a possible rare edge case here: the attachment_id comes from the image block (hard-coded in post_content), but if the site was moved, i.e. posts have been exported and imported at in another site, the post IDs may change. Seems this will also need to confirm that `$image_meta` is for the same image. (Not in scope here, just need to remember at add it).